src/test/components/NetworkChart.test.js

```diff
@@ -124,27 +134,33 @@ describe('NetworkChartRowBar phase calculations', function() {
         pri: 20,
         count: 10,
         status: 'STATUS_REDIRECT',
-        connectStart: 20,
         startTime: 10,
-        endTime: 90,
-        requestStart: 20,
+        // With an endTime at 99, the profile range goes until 100, which
+        // gives integer values for test results.
+        endTime: 99,
+        domainLookupStart: 20,
+        domainLookupEnd: 24,
+        connectStart: 25,
+        tcpConnectEnd: 26,
+        secureConnectionStart: 26,
+        connectEnd: 28,
+        requestStart: 30,
         responseStart: 60,
         responseEnd: 80,
       }
     );
-    expect(styleForClass('.networkChartRowItemBarRequestQueue')).toMatch(
-      'width: 12.5%'
-    );
-    expect(styleForClass('.networkChartRowItemBarRequest')).toMatch(
-      'width: 50%'
-    );
-    expect(styleForClass('.networkChartRowItemBarResponse')).toMatch(
-      'width: 25%'
-    );
+
+    expect(getPhaseElementStyles()).toEqual([
+      'left: 0px; width: 20px; opacity: 0;',
+      'left: 20px; width: 20px; opacity: 0.3333333333333333;',
+      'left: 40px; width: 60px; opacity: 0.6666666666666666;',
+      'left: 100px; width: 40px; opacity: 1;',
+      'left: 140px; width: 38px; opacity: 0;',
+    ]);
```

Comment:
I'm not completely happy about this way of testing, but this does the job properly and is simple.